### PR TITLE
Adding share modal for socials media

### PR DESCRIPTION
**Pull request by codiejay**
image 

Adding the share button to the list of clickable bttns on wikiAction tab. 

image 

Modal pops when share button is clicked and users can decide to Click on the Copy URL bttn or click on any of the socials icon to share to that respective platform. 

closes #520, closes #520, closes https://github.com/EveripediaNetwork/issues/issues/520

### CODE REVIEW

**Review by Softdev1 — APPROVED** [review]
Awesome one. well done @codiejay

File: src/components/Wiki/WikiPage/CustomModals/ShareWikiModal.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+]
+
+const ShareWikiModal = ({
+  onClose = () => {},
+  isOpen = false,
+  ...rest
+}: Partial<ModalProps>) => {
+  const router = useRouter()
+  const { hasCopied, onCopy } = useClipboard(
+    `https://alpha.everipedia.org${router.asPath}`,
```
Comment (kesar) [inline src/components/Wiki/WikiPage/CustomModals/ShareWikiModal.tsx:66]: needs to be from config

File: src/components/Wiki/WikiPage/CustomModals/ShareWikiModal.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+              p="0.5rem"
+              bg="brand.150"
+              _dark={{ bg: '#232934' }}
+            >
+              <Link
+                whiteSpace="nowrap"
+                href={router.asPath}
+                color="#2D3748"
+                _dark={{ color: '#b1b2b5' }}
+              >{`https://alpha.everipedia.org${router.asPath}`}</Link>
```
Comment (kesar) [inline src/components/Wiki/WikiPage/CustomModals/ShareWikiModal.tsx:101]: same

File: src/components/Wiki/WikiPage/CustomModals/ShareWikiModal.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+          <Flex fontWeight="bold" mt="3rem" flexDirection="column">
+            <Text color="#1A202C" fontSize="sm" _dark={{ color: 'white' }}>
+              Or share via:
+            </Text>
+            <Wrap mt="1rem">
+              {SHARING_OPTIONS.map(item => {
+                return (
+                  <item.label
+                    url={`https://alpha.everipedia.org${router.asPath}`}
```
Comment (kesar) [inline src/components/Wiki/WikiPage/CustomModals/ShareWikiModal.tsx:113]: same

**Comment by kesar** [comment]
does it look fine on mobile?

**Comment by Softdev1** [comment]
on mobile it looks cool actually, I doubled checked on my mobile device

**Review by kesar — APPROVED** [review]

**Pull request merged**